fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -2629,4 +2649,17 @@ private static Map<SlotRef, Expr> getExpectedBinaryPredicates(
             return null;
         }
     }
+
+    private boolean isAddDefaultOrderBy() {
+        if (ConnectContext.get() == null
+                || ConnectContext.get().getSessionVariable() == null
+                || !ConnectContext.get().getSessionVariable().isEnableDefaultOrder()) {
+            return false;
```

Comment:
u should check the `TableRef` has no `lateralViewRefs`

Reply:
check only one TableRef already cover the case. We cannot use only literal view at from clause.